### ISSUE TITLE
SystemVerilog: Compiler incorrectly rejects multiple procedural assignments to a variable bit/part when another bit/part of that variable is driven by a continuous assignment.

### ISSUE THREAD

**Issue by qian-gu** [steveicarus/iverilog#703]
Hi, I get a error message saying `error: Theses bit are already driven.`  when comiple following code:

```systemverilog
logic [7 : 0] dout;
logic [7 : 0] sel;

for (genvar i = 0; i < 8; i++) begin
  if (i == 0) begin

    assign dout[i] = 1'b0;

  end else if (i == 1) begin

    assign dout[i] = 1'b1;

  end else begin

    // using always block reports error
    always @(*) begin
        if (sel[i]) dout[i] = 1'b1;
        else dout[i] = 1'b0;
    end

  end
end
```

But if I replace the always block with assign statement, the error message disapears,

```systemverilog
logic [7 : 0] dout;
logic [7 : 0] sel;

for (genvar i = 0; i < 8; i++) begin
  if (i == 0) begin

    assign dout[i] = 1'b0;

  end else if (i == 1) begin

    assign dout[i] = 1'b1;

  end else begin

    assign dout[i] = sel[i] ? 1'b1 : 1'b0;

  end
end
```

so is this a bug or intend result of some consideration?

**Comment by martinwhitaker** [comment]
It's a bug. A simpler test case is
````
module test();

logic       a;
logic [1:0] v;

assign v[0] = 1'b0;

always @* begin
  if (a)
    v[1] = 1'b1;
  else
    v[1] = 1'b0;
end

endmodule
````
Because `v` is partially driven by a continuous assignment, the compiler is checking that it is not driven by multiple assignments. But that should only be done for continuous assignments, not for procedural assignments..

**Comment by martinwhitaker** [comment]
This was fixed by commit d4759e02aaf88a920ee48849056e7d133a44f282.
